### PR TITLE
Cc feature/17/subscribe to a post

### LINKED ISSUE #15: Unsubscribe to a User's Posts
As a reader I would like to unsubscribe from an author's Posts so that I won't see the Posts of author's I am no longer interested in reading.

**Given** a user is viewing the Post details of an author they are subscribed to  
**When** they select the option to unsubscribe to the author  
**Then** the Subscription end datetime should be updated with the current datetime  
**And** the system should behave as it did before the user had created the Subscription

### LINKED ISSUE #17: Subscribe to a User's Posts
As a reader I would like to be able to subscribe to authors that I enjoy so that I can more easily find new Posts they have written.

**Given** A user is on a Post detail page  
**When** they select the option to subscribe to the Post's author's other Posts  
**Then** a Subscription should be created in the database  
**And** the Subscription's begin datetime should be set to the current time

### PR DESCRIPTION
**Pull request by ccote95**
### Purpose 
To implement the ability to subscribe and unsubscribe

### Files Changed 
- Created a new DTO for creating a subscription in `Models/DTOs/Subscription/SubscriptionCreate.cs`
-  Created a manager to handle subscribing and unsubscribe in `client/src/managers/subscriptionManager.js`
- Created controllers to handle subscribing and unsubscribe in `Controllers/SubscribeController.cs`
- Wrote logic to handle rendering a subscribe or unsubscribe from a post in `client/src/components/posts/PostDetails.jsx`

### Testing
Fetch, setup, and run according to the project README. but do not create a new migration
Confirm the following:
- As a user, when viewing a posts details if you are not the author of the post and not subscribed to them a subscribe button will render
- If you choose to subscribe to that post, when you click on the subscribe button it will create a new subscription and now render the button as unsubscribe
- If you choose to unsubscribe from the post, when you click unsubscribe it will remove the subscription and render the button back to subscribe

closes #17 
closes #15

### CODE REVIEW

**Review by joshuahamrick — APPROVED** [review]

File: client/src/managers/subscriptionManager.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        },
+        body: JSON.stringify(newSub)
+    })
+}
+
+export const getSubscriptionsById = (Id) => {
+    return fetch(`${_apiUrl}/${Id}`).then(res => res.json())
+};
+
+export const unsubscribe = (creatorId, followerId) => {
```
Comment (ztrouy) [inline client/src/managers/subscriptionManager.js:17]: Change name to match createSubscription

File: client/src/components/posts/PostDetails.jsx [hunk 2]
```diff
@@ -46,6 +49,22 @@ export const PostDetails = ({ loggedInUser }) => {
...
+            creatorId: authorId,
+            followerId: followerId
+        }
+        createSubscription(newSubscription)
+        .then(() =>  getSubscriptionsById(loggedInUser.id)
+            .then(setUserSubscriptions))
+    }
+
+    const handleUnsubscribe = (CreatorId, followerId) => {
+        unsubscribe(CreatorId, followerId)
```
Comment (ztrouy) [inline client/src/components/posts/PostDetails.jsx:63]: Change to use new name

**Review by ztrouy — CHANGES_REQUESTED** [review]

File: client/src/components/posts/PostDetails.jsx [hunk 3]
```diff
@@ -46,6 +49,22 @@ export const PostDetails = ({ loggedInUser }) => {
         })
     }
 
+    const handleSubscribe= (authorId, followerId) => {
+        const newSubscription = {
+            creatorId: authorId,
+            followerId: followerId
+        }
+        createSubscription(newSubscription)
```
Comment (ztrouy) [inline client/src/components/posts/PostDetails.jsx:57]: Fix formatting

File: client/src/components/posts/PostDetails.jsx [hunk 4]
```diff
@@ -46,6 +49,22 @@ export const PostDetails = ({ loggedInUser }) => {
...
+        .then(() =>  getSubscriptionsById(loggedInUser.id)
+            .then(setUserSubscriptions))
+    }
+
+    const handleUnsubscribe = (CreatorId, followerId) => {
+        unsubscribe(CreatorId, followerId)
+        .then(() =>  getSubscriptionsById(loggedInUser.id)
```
Comment (ztrouy) [inline client/src/components/posts/PostDetails.jsx:64]: Fix formatting

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by ztrouy — APPROVED** [review]
:shipit:

**Pull request merged**